code/game/turfs/turf.dm

```diff
@@ -129,6 +129,8 @@
 
 /turf/attack_hand(mob/user)
 	. = ..()
+	if(SEND_SIGNAL(user, COMSIG_MOB_ATTACK_HAND_TURF, src) & COMPONENT_NO_ATTACK_HAND)
```

Comment:
This is potentially wasteful but since its only on turf attack_hand its actually mostly okay.